### PR TITLE
Adds `intialDate` to `MacosDatePicker`

### LINKED ISSUE #310: add `initialDate` to `MacosDatePicker`
## Use case
I'd like to control the initial date set in `MacosDatePicker` widget.

## Proposal

Create a parameter `initialDate` that sets `_MacosDatePickerState._initialDate` on init state. If the value of the parameter `initialDate` is `null`, then use `DateTime.now()`.

Example of ideal usage:

```dart
MacosDatePicker(
  style: DatePickerStyle.textual,
  initialDate: DateTime(2022, 10, 10),
  onDateChanged: _onDateChanged,
)
```

### PR DESCRIPTION
**Pull request by Luckey-Elijah**
Closes #310 

## Pre-launch Checklist

- [X] I have run `dartfmt` on all changed files 
- [X] I have incremented the package version as appropriate and updated `CHANGELOG.md` with my changes 
- [X] I have added/updated relevant documentation 
- [X] I have run "optimize/organize imports" on all changed files
- [X] I have addressed all analyzer warnings as best I could

### CODE REVIEW

**Comment by Luckey-Elijah** [comment]
@GroovinChip ready for review.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+## [1.7.7]
+* Add's optional `initialDate` to `MacosDatePicker`
```
Comment (GroovinChip) [inline CHANGELOG.md]:
```suggestion
* Adds optional `initialDate` to `MacosDatePicker`
```

File: lib/src/selectors/date_picker.dart [hunk 2]
```diff
@@ -53,12 +54,18 @@ class MacosDatePicker extends StatefulWidget {
   /// {macro onDateChanged}
   final OnDateChanged onDateChanged;
 
+  /// Set an intial date for the picker.
+  /// Defaults to `DateTime.now()`.
```
Comment (GroovinChip) [inline lib/src/selectors/date_picker.dart]:
```suggestion
  /// Set an intial date for the picker.
  ///
  /// Defaults to `DateTime.now()`.
```

File: lib/src/selectors/date_picker.dart [hunk 3]
```diff
@@ -53,12 +54,18 @@ class MacosDatePicker extends StatefulWidget {
...
+  final DateTime? initialDate;
+
   @override
   State<MacosDatePicker> createState() => _MacosDatePickerState();
 }
 
 class _MacosDatePickerState extends State<MacosDatePicker> {
-  final _initialDate = DateTime.now();
+  final _today = DateTime.now();
+  late final _initialDate = widget.initialDate ?? _today;
```
Comment (GroovinChip) [inline lib/src/selectors/date_picker.dart:68]: Is there a reason that `_initialDate` needs to be `late`?

**Review by GroovinChip — CHANGES_REQUESTED** [review]

File: lib/src/selectors/date_picker.dart (hunk 3 above)
Comment (Luckey-Elijah) [reply lib/src/selectors/date_picker.dart:68]: Yes, `widget` and `_today` can't be accessed in an initializer, but can be with `late` applied. Another way to write it is with an initialization happening in `initState`

**Comment by Luckey-Elijah** [comment]
@GroovinChip please see with latest changes pulled from master. Should be ready for tests + merge if no other review suggestions

**Review by GroovinChip — APPROVED** [review]
LGTM! Thanks for the contribution 🎉

**Pull request merged**